tests/test_gbdesAstrometricFit.py

```diff
@@ -540,6 +570,285 @@ def test_run(self):
+            starDecs = np.random.random(cls.nStars) * (decMax - decMin) + decMin
+            # Remove neighbors:
+            with Matcher(starRAs, starDecs) as matcher:
+                idx = matcher.query_groups(cls.task.config.matchRadius / 3600.0, min_match=2)
+            if len(idx) > 0:
+                neighbors = np.unique(np.concatenate(idx))
+                starRAs = np.delete(starRAs, neighbors)
+                starDecs = np.delete(starDecs, neighbors)
+                cls.nStars = len(starRAs)
+                starIds = np.arange(cls.nStars)
+
+            allStarIds.append(starIds)
+            allStarRAs.append(starRAs)
+            allStarDecs.append(starDecs)
+
+            corners = [
+                lsst.geom.SpherePoint(ra, dec, lsst.geom.degrees).getVector()
+                for (ra, dec) in zip([raMin, raMax, raMax, raMin], [decMin, decMin, decMax, decMax])
+            ]
+            conv_box = lsst.sphgeom.ConvexPolygon.convexHull(corners)
+            # Make a reference catalog that will be loaded into
+            # ReferenceObjectLoader
+            refDataId, deferredRefCat = cls._make_refCat(
+                starIds, starRAs, starDecs, inReferenceFraction, conv_box
+            )
+            refDataIds.append(refDataId)
+            deferredRefCats.append(deferredRefCat)
+
+        cls.refObjectLoader = ReferenceObjectLoader(refDataIds, deferredRefCats)
+        cls.refObjectLoader.config.requireProperMotion = False
+        cls.refObjectLoader.config.anyFilterMapsToThis = "test_filter"
+
+        cls.task.refObjectLoader = cls.refObjectLoader
+
+        allRefObjects, allRefCovariances = {}, {}
+        for f, fieldRegion in cls.fieldRegions.items():
+            refObjects, refCovariance = cls.task._load_refcat(
+                cls.refObjectLoader, cls.extensionInfo, epoch=cls.exposureInfo.medianEpoch, region=fieldRegion
+            )
+            allRefObjects[f] = refObjects
+            allRefCovariances[f] = refCovariance
+        cls.refObjects = allRefObjects
+
+        # Get True WCS for stars:
+        with open(os.path.join(cls.datadir, "sample_global_wcs.yaml"), "r") as f:
+            cls.trueModel = yaml.load(f, Loader=yaml.Loader)
+
+        cls.trueWCSs = cls._make_wcs(cls.trueModel, cls.inputVisitSummary)
+
+        cls.isolatedStarCatalogs, cls.isolatedStarSources = cls._make_isolatedStars(
+            allStarIds, allStarRAs, allStarDecs, cls.trueWCSs, inScienceFraction
+        )
+
+        cls.outputs = cls.task.run(
+            cls.inputVisitSummary,
+            cls.isolatedStarSources,
+            cls.isolatedStarCatalogs,
+            instrumentName=cls.instrumentName,
+            refEpoch=cls.refEpoch,
+            refObjectLoader=cls.refObjectLoader,
+        )
+
+    @classmethod
+    def _make_isolatedStars(cls, allStarIds, allStarRAs, allStarDecs, trueWCSs, inScienceFraction):
+        # Take a subset of the simulated data
```

Comment:
I'm having trouble understanding what is meant by "simulated data" for these tests and how these are used vs. the real data you are also using (so it's hard to get a feel for how much of the task is getting tested).  A few words of explanation would go a long way (for me at least!)

Reply:
I think these were notes to myself that I missed converting into docs. I added full documentation here, plus a short explanation in the class documentation for the parent class: 
```
    This class tests `GbdesAstrometricFit` using real `visitSummaryTable`s
    from HSC RC2 processing, with simulated sources for those visits.
```
The only real data in the tests is the `visitSummaries`. The sources and true object positions are all simulated from some "true WCS", which is known to be realistic given the input information in the `visitSummaries`. The overarching test of the task, which happens in `test_run()`,  checks that, given the `inputSummaries` and simulated source catalogs, you can recover the "trueWCS" to a certain accuracy.